in_toto/verifylib.py

```diff
@@ -1306,6 +1326,42 @@ def verify_threshold_constraints(layout, chain_link_dict):
                     step_name=step.name, keyid=keyid)))
 
 
+def get_traceback_info():
+  """
+  <Purpose>
+    Retrieves traceback information for the error message meant to be printed 
+    out when a RuleVerificationError is raised using elements of the rule trace.
+
+  <Arguments>
+    None.
+
+  <Exceptions>
+    None.
+
+  <Side Effects>
+    None.
+
+  <Returns>
+    A string containing a traceback for all processed rules with rule names 
+    and the states of the queues after each round of rule processing.
+
+  """
+
+  traceback_str = "Trace for earlier '{0}' rules of item '{1}':\n".format(
+      RULE_TRACE["type"], RULE_TRACE["name"])
+
+  traceback_str += "Initial state of queues:\n"
+  traceback_str += pprint.pformat(RULE_TRACE["trace"].pop(0)) + "\n"
+
+  for trace_entry in RULE_TRACE["trace"]:
+    traceback_str += "Queues after processing of rule '{0}':\n".format(
+        trace_entry["rule"])
+    queues = {k: trace_entry[k] for k in ('artifacts', 'materials', 'products')}
```

Comment:
Our style guidelines [discourage the use of dict comprehensions](https://github.com/secure-systems-lab/code-style-guidelines#no-lambda-functions-or-lisp-esque-code). Personally, I think they are really neat, but let's not use them.

What about a second regular for loop, e.g....?

```python
for name in ["artifacts", "materials", "products"]:
  traceback_str += "\t{}: {}\n".format(name, trace_entry[name])
```
You could format the "initial state of queues" in that for loop too.  Then you wouldn't need `pprint` anymore.  As an alternative to `pop`ing the first line you can use an index from `enumerate` (remember?) so that you can distinguish between the first and the rest of the trace lines.

If you do use `pop`, please mention it in the docstring's side effects.